### DISCUSSION TITLE
On Clicking EditAction Button in table show notiification based on certian condition otherwise redirect it to edit form

### DISCUSSION THREAD

**Discussion by Muhammad-fareed** [filamentphp/filament#11701] [Help]
### Package

Actions

### Package Version

3.0.0

### How can we help you?

Hello! i want to check a specific condition like if my order status is dispatched and user clicks on editAction button it will not redirect it to edit form rather it should return a message like : the order is dispatched unable to edit . I tried to use ->before() but its not working. If you know any other way please let me know.

**Comment by leandrocfe** [comment]
Disabled?

```php
Tables\Actions\EditAction::make()
    ->disabled(fn (Model $record): bool => $record->dispatched)
```

**Reply by Muhammad-fareed** [reply]
No I want to show pop up on clicking edit showing user a message: "unable to edit".

**Comment by dissto** [comment]
What exactly wasn't working using before? And what prevents the user from manually going to the edit page?

**Reply by Muhammad-fareed** [reply]
there are different stages in my app like odering packing and dispatching i want the user to not go to edit page when the order is dispatched so i want to do additionall thing like showing popup that user is unable to edit as the order is dispatched when user click on edit button. For now i am doing this
 `Tables\Actions\EditAction::make()
                    ->visible(fn (Order $record) => $record->order_status == "dispatched" ? false : true),`
but i am unable to achieve the popup llike thing in filament

**Reply by micraux** [reply]
You can create another action, like this:
```php
Tables\Actions\EditAction::make()
    ->visible(fn (Order $record) => $record->order_status != "dispatched"),

Action::make('edit')
    ->visible(fn (Order $record) => $record->order_status == "dispatched"),
    ->requiresConfirmation()
    ->modalHeading('Edit')
    ->modalDescription('Unable to edit')
    ->modalSubmitActionLabel('Ok')
```
